### PR TITLE
Add support for ssd16xx monochrome EPD controllers

### PR DESCRIPTION
**Pull request by aurel32**
This PR updates the existing ssd1673 driver to add support for more EPD controllers. It has been tested with a Waveshare 1.54inch e-Paper Module using a ssd1607 controller. Note however that I used an additional patch to update the LUTs, as this EPD use different ones. I still have to find a way to provide an easy way to use different LUTs for different EPD.

I believe this driver can be extended to provide support for all the ssd16xx family, including the 3- or 4- colors ones. It therefore might make sense to rename this driver.

### CODE REVIEW

File: drivers/display/ssd1673.c [hunk 1]
```diff
@@ -432,9 +465,9 @@ static int ssd1673_controller_init(struct device *dev)
 	}
 	ssd1673_busy_wait(driver);
 
-	tmp[0] = SSD1673_PANEL_LAST_GATE;
-	tmp[1] = 0U;
-	err = ssd1673_write_cmd(driver, SSD1673_CMD_GDO_CTRL, tmp, 2);
+	len = push_y_param(tmp, SSD1673_PANEL_LAST_GATE);
+	tmp[len] = 0U; len++;
```
Comment (vanwinkeljan) [inline drivers/display/ssd1673.c]: Small remark move ```len++``` to a new line

**Review by jfischer-no — CHANGES_REQUESTED** [review]
looks good, tested, just few comments

File: boards/arm/reel_board/reel_board.dts [hunk 2]
```diff
@@ -123,6 +123,8 @@
 		reg = <0>;
 		width = <250>;
 		height = <120>;
+		width-bits = <8>;
```
[the remark this answers is not in the document]
Comment (aurel32) [reply boards/arm/reel_board/reel_board.dts]: I am fine changing it, but I think we should keep `bits` in the name, that's the quantity represented by this value.

File: drivers/display/ssd1673.c [hunk 3]
```diff
@@ -99,22 +99,52 @@ static inline void ssd1673_busy_wait(struct ssd1673_data *driver)
...
+#endif
+}
+
+static inline size_t push_y_param(u8_t *data, u16_t y)
+{
+#if DT_SOLOMON_SSD1673FB_0_HEIGHT_BITS == 8
+	data[0] = y & 0xff;
+	return 1;
+#elif DT_SOLOMON_SSD1673FB_0_HEIGHT_BITS == 16
+	data[0] = y & 0xff;
```
[the remark this answers is not in the document]
Comment (aurel32) [reply drivers/display/ssd1673.c]: Good idea, i'll fix that

File: drivers/display/ssd1673.c (hunk 1 above)
Comment (aurel32) [reply drivers/display/ssd1673.c]: Good point i'll fix that

**Review by vanwinkeljan — APPROVED** [review]

**Comment by aurel32** [comment]
recheck

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**